src/Paprika/Merkle/ComputeMerkleBehavior.cs

```diff
@@ -573,15 +688,37 @@ private void EncodeBranch(scoped in Key key, scoped in ComputeContext ctx, scope
             var budget = ctx.Budget;
 
             // parallel calculation
-            ParallelUnbalancedWork.For(0, NibbleSet.NibbleCount, s_parallelOptions, nibble =>
+#if SNAP_SYNC_SUPPORT
+            //Unpack RlpMemo to be used in parallel calculations
+            //used for node proofs in snap sync
+            Keccak[] memoKeccaks = new Keccak[NibbleSet.NibbleCount];
+            if (memoize)
             {
-                var childPath = NibblePath.Single((byte)nibble, 0);
+                for (byte i = 0; i < NibbleSet.NibbleCount; i++)
+                {
+                    if (memo.TryGetKeccak(i, out var keccakBytes))
+                        memoKeccaks[i] = new Keccak(keccakBytes);
+                }
+            }
+#endif
 
+            Parallel.For((long)0, NibbleSet.NibbleCount, s_parallelOptions, nibble =>
```

Comment:
This needs to be made back to `ParallelUnbalancedWork`

Reply:
Done